### PR TITLE
FASTQ to FASTA conversion

### LINKED ISSUE #299: Add a script for converting FASTQ to FASTA
The only slight complication here is that the FASTA sequence can span multiple lines. We can read it using our own tools (which use BioPython), which will handle this.

### PR DESCRIPTION
**Pull request by terrycojones**
Fixes #299

### CODE REVIEW

File: bin/convert-fastq-to-fasta.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+from dark.fastq import FastqReads
+
+# Print a usage message if any arguments were given on the command
+# line. This is to remind people who want to provide filenames that they
+# must use '<' to supply our stdin and '>' to store our output.
+
+if len(sys.argv) > 1:
+    print >>sys.stderr, (
+        'Usage: %s <input.fastq [>output.fasta]' % basename(sys.argv[0]))
```
Comment (bamueh) [inline bin/convert-fastq-to-fasta.py]: ```'Usage: %s < input.fastq [> output.fasta]'```

File: test/test_fastq.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+from mock import patch
+from mocking import mockOpen
+
+
+class TestFastqReads(TestCase):
+    """Tests for the L{dark.fastq.FastqReads} class."""
+
+    def testEmpty(self):
+        """An empty FASTQ file results in an empty iterator."""
+        mockOpener = mockOpen()
```
Comment (bamueh) [inline test/test_fastq.py]: Why are the quotes on the same line?

File: bin/convert-fastq-to-fasta.py (hunk 1 above)
Comment (terrycojones) [inline bin/convert-fastq-to-fasta.py]: Fixed.

File: test/test_fastq.py (hunk 2 above)
Comment (terrycojones) [inline test/test_fastq.py]: Because I copied `test_fasta.py`. I just fixed both files.

**Pull request merged**